llvm/test/Transforms/InstCombine/ashr-icmp-minmax-idiom-break.ll

```diff
@@ -10,7 +10,8 @@ define i64 @dont_break_minmax_i64(i64 %conv, i64 %conv2) {
 ; CHECK-SAME: (i64 [[CONV:%.*]], i64 [[CONV2:%.*]]) {
 ; CHECK-NEXT:    [[MUL:%.*]] = mul nsw i64 [[CONV]], [[CONV2]]
 ; CHECK-NEXT:    [[SHR:%.*]] = ashr i64 [[MUL]], 4
-; CHECK-NEXT:    [[SPEC_SELECT_I:%.*]] = call i64 @llvm.smin.i64(i64 [[SHR]], i64 348731)
+; CHECK-NEXT:    [[CMP4_I:%.*]] = icmp slt i64 [[MUL]], 5579712
```

Comment:
It breaks the min/max idiom.

Reply:
You could try passing `matchSelectPattern` result to the function and only drop the 1 use check if the result is `SPF_UNKNOWN`

Reply:
Are you referring to this kind of usage: `matchSelectPattern(Shr->user_back(), Lhs, Rhs).Flavor == SPF_UNKNOWN`?

Reply:
Yes

Reply:
When I executed `matchSelectPattern(Shr->user_back(), Lhs, Rhs).Flavor == SPF_UNKNOWN` within the function, it produced an `SPF_UNKNOWN` output for the `dont_break_minmax` test on this code snippet.

Reply:
Hmm, I would have assumed that would indicate we are breaking the selection idiom. Would have to spend some time digging to see where it goes wrong. maybe @dtcxzyw has an idea?

Reply:
The check already exists here:
https://github.com/llvm/llvm-project/blob/0c89427b99f6f6d7c217c70ff880ca097340f9a4/llvm/lib/Transforms/InstCombine/InstCombineCompares.cpp#L7043-L7058

Could you please move the call to `foldICmpWithConstant` after the check?
Related patch: https://github.com/llvm/llvm-project/pull/82472

Reply:
> Failed Tests (3):
  LLVM :: Transforms/PhaseOrdering/ARM/arm_mult_q15.ll
  LLVM :: Transforms/PhaseOrdering/icmp-ashr-breaking-select-idiom.ll
  LLVM :: Transforms/PhaseOrdering/loop-access-checks.ll

@SahilPatidar Please update these tests.